java/code/webapp/WEB-INF/decorators/layout_head.jsp

```diff
@@ -62,14 +62,17 @@
         </c:otherwise>
     </c:choose>
 
+    <!-- expose user preferred language to the application -->
+    <c:set var="currentLocale" value="${GlobalInstanceHolder.MENU_TREE.getCurrentLocale(pageContext)}"/>
+    <script> preferredLocale='${currentLocale}'</script>
```

Comment:
Can we put this into `window.preferredLocale`?

Reply:
Yes of course we can. I will change it